tests/testthat/test-extract_fin_year.R

```diff
@@ -22,3 +22,85 @@ test_that("Non-date formats produce an error", {
 test_that("NAs are handled correctly", {
   expect_equal(extract_fin_year(c(lubridate::dmy(05012020), NA)), c("2019/20", NA))
 })
+
+test_that("YYYY/YY format applied correctly", {
+  expect_equal(extract_fin_year(c(lubridate::dmy(NA, "01/April/1999"), NA)),
+               c(NA, "1999/00", NA))
+  expect_equal(extract_fin_year(c(lubridate::dmy(NA, "01/April/2000"), NA)),
+               c(NA, "2000/01", NA))
+  expect_equal(extract_fin_year(c(lubridate::dmy(NA, "01/April/2001"), NA)),
+               c(NA, "2001/02", NA))
+  expect_equal(extract_fin_year(c(lubridate::dmy(NA, "31/March/1999"), NA)),
+               c(NA, "1998/99", NA))
+  expect_equal(extract_fin_year(c(lubridate::dmy(NA, "31/March/2000"), NA)),
+               c(NA, "1999/00", NA))
+  expect_equal(extract_fin_year(c(lubridate::dmy(NA, "31/March/2001"), NA)),
+               c(NA, "2000/01", NA))
+  expect_equal(extract_fin_year(c(lubridate::dmy(NA, "01/December/1999"), NA)),
+               c(NA, "1999/00", NA))
+  expect_equal(extract_fin_year(c(lubridate::dmy(NA, "01/December/2000"), NA)),
+               c(NA, "2000/01", NA))
+  expect_equal(extract_fin_year(c(lubridate::dmy(NA, "01/December/2999"), NA)),
+               c(NA, "2999/00", NA))
+  expect_equal(extract_fin_year(c(lubridate::dmy(NA, "01/December/3000"), NA)),
+               c(NA, "3000/01", NA))
+
+  expect_equal(extract_fin_year(
+    lubridate::as_datetime(
+      c(lubridate::dmy(NA, "01/April/1999"), NA)
+    )
+  ),
+  c(NA, "1999/00", NA))
+
+  expect_equal(extract_fin_year(
+    lubridate::as_datetime(
+      c(lubridate::dmy(NA, "01/December/2000"), NA)
+    )
+  ),
+  c(NA, "2000/01", NA))
+
+  expect_equal(extract_fin_year(
+    lubridate::as_datetime(
+      c(lubridate::dmy(NA, "01/April/0001"), NA)
+    )
+  ),
+  c(NA, "0001/02", NA))
+})
+
+test_that("Correct outputs", {
```

Comment:
This test would be clearer as a snapshot I think.

1. Setup the input data frame.
2. Use test_snapshot(extract_fin_year(input_data)).
3. Run the tests devtools::test()
4. Check and accept the snapshot.